engine/src/main/java/org/hibernate/validator/internal/engine/cascading/LegacyMapValueExtractor.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine.cascading;
+
+import java.util.Map;
+
+import org.hibernate.validator.spi.cascading.ValueExtractor;
+
+class LegacyMapValueExtractor implements ValueExtractor<Map<?, ?>> {
```

Comment:
Currently it's not needed, but I'm wondering whether we should require `@ExtractedValue` on the actual type itself:

    class LegacyMapValueExtractor implements ValueExtractor<@ExtractedValue Map<?, ?>> { ... }

It seems nice for consistency purposes and makes this extractor's purpose more apparent.